controller/modules/macro/step.go

```diff
@@ -61,6 +70,47 @@ func (s *Step) Run(c controller.Controller, reverse bool) error {
 		}
 		log.Println("macro-subsystem: executing step: sleep for", int(w.Duration), "seconds")
 		time.Sleep(w.Duration * time.Second)
+		return nil
+	case "waitfortrigger":
+		var w WaitForTriggerStep
+		if err := json.Unmarshal(s.Config, &w); err != nil {
+			return err
+		}
+
+		//Set up a way for the step to wait unil the goroutine completes
+		var wg sync.WaitGroup
+
+		//Set up a subscription for the metric
+		ctx, cancelFn := context.WithCancel(context.Background())
+		messages, err := c.PubSub().Subscribe(ctx, w.Metric)
+		if err != nil {
+			panic(err)
+		}
+
+		//spin a goroutine to monitor the emitted metric
+		go func() {
+			for msg := range messages {
+				msg.Ack()
+
+				formula := "metric " + w.Operator + " " + w.Operand
```

Comment:
I agree that the macro step implementation needs a lot of work.  I put this in only to show a way to use the message bus, including the ability to unsubscribe.  I think the type safety is going to be a challenge without some extensive UI and backend work.  

Here are a few ideas.  I'm leaning toward the second one so far.

- Maybe we could add a struct/map/some kind of look up that lists all topics that can be published with metadata about the payload. 
- **Publishers register metadata about their payload with the message bus**
- Add a Publisher interface to the modules that includes a function to provide metadata about the topics it will publish